org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semantictokens/SemanticTokensService.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) Avaloq Evolution AG
+ * Schwerzistrasse 6, 8807 Freienbach, Switzerland, http://www.avaloq.com
+ * All Rights Reserved.
+ *
+ * This software is the confidential and proprietary information of Avaloq Evolution AG.
+ * You shall not disclose whole or parts of it and shall use it only in accordance with the terms of the
+ * licence agreement you entered into with Avaloq Evolution AG.
+ */
+
+package org.eclipse.xtext.ide.server.semantictokens;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.SemanticTokens;
+import org.eclipse.lsp4j.SemanticTokensParams;
+import org.eclipse.lsp4j.SemanticTokensRangeParams;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.eclipse.xtext.ide.editor.syntaxcoloring.HighlightingStyles;
+import org.eclipse.xtext.ide.editor.syntaxcoloring.ISemanticHighlightingCalculator;
+import org.eclipse.xtext.ide.editor.syntaxcoloring.LightweightPosition;
+import org.eclipse.xtext.ide.editor.syntaxcoloring.MergingHighlightedPositionAcceptor;
+import org.eclipse.xtext.ide.server.Document;
+import org.eclipse.xtext.ide.server.UriExtensions;
+import org.eclipse.xtext.resource.IResourceServiceProvider;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.util.CancelIndicator;
+
+import com.google.inject.Inject;
+
+/**
+ * The Class SemanticTokensService.
+ */
+public class SemanticTokensService {
+	@Inject
+	private UriExtensions uriExtensions;
+	@Inject
+	private IResourceServiceProvider.Registry languagesRegistry;
+	private List<String> tokenTypes;
+
+	SemanticTokensService() {
+		this.tokenTypes = new ArrayList<>();
```

Comment:
i guess this one should go to an extra method

Reply:
thanks. Done